datadog/src/lib.rs

```diff
@@ -0,0 +1,100 @@
+use hyper::{Body, Client, Method, Request};
+use hyper_tls::HttpsConnector;
+use serde::{ser, Serialize};
+use std::collections::BTreeMap as Map;
+
+#[derive(Debug, Serialize)]
+pub struct Event {
+    pub ddsource: String,
+    pub service: String,
+    #[serde(serialize_with = "serialize_ddtags")]
+    pub ddtags: DdTags,
+    pub hostname: String,
+    pub message: String,
+}
+
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+pub struct Error {
+    pub code: u16,
+}
+
+pub type DdTags = Map<String, String>;
+
+fn serialize_ddtags<S>(ddtags: &DdTags, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: ser::Serializer,
+{
+    ddtags
+        .iter()
+        .map(|(k, v)| [k.clone(), v.clone()].join(":"))
+        .collect::<Vec<_>>()
+        .join(",")
+        .serialize(serializer)
+}
+
+pub async fn send_event<T>(value: &T, dd_api_key: String) -> Result<(), Error>
```

Comment:
Is there a benefit for `send_event` being generic here?

It seems like the method is really only intended to be passed in Event structs.

Reply:
this is a months old WIP. not ready for review, let you know after connecting to mina.

Reply:
Sorry. I was just taking getting a sense of what was going on.

Reply:
Made it use the Event struct.